src/pages/studyView/tabs/SummaryTab.tsx

```diff
@@ -521,237 +612,266 @@ export class StudySummaryTab extends React.Component<
-                    props.chartMeta!.uniqueKey
-                );
+                return props;
+            },
+            [ChartTypeEnum.SCATTER]: () => {
                 const chartInfo = this.store.getXvsYScatterChartInfo(
-                    props.chartMeta!.uniqueKey
+                    chartMeta.uniqueKey
                 )!;
                 const settings = this.store.getXvsYChartSettings(
-                    props.chartMeta!.uniqueKey
+                    chartMeta.uniqueKey
                 )!;
-                props.title = props.chartMeta!.displayName;
-                props.promise = this.store.clinicalDataDensityCache.get({
-                    chartInfo,
-                    xAxisLogScale: !!settings.xLogScale,
-                    yAxisLogScale: !!settings.yLogScale,
-                });
-                props.showLogScaleXToggle = logScalePossible(
-                    chartInfo.xAttr.clinicalAttributeId
-                );
-                props.showLogScaleYToggle = logScalePossible(
-                    chartInfo.yAttr.clinicalAttributeId
-                );
-                props.logScaleXChecked = settings.xLogScale;
-                props.logScaleYChecked = settings.yLogScale;
-                props.onToggleLogScaleX = () => {
-                    const settings = this.store.getXvsYChartSettings(
-                        props.chartMeta!.uniqueKey
-                    )!;
-                    settings.xLogScale = !settings.xLogScale;
-                };
-                props.onToggleLogScaleY = () => {
-                    const settings = this.store.getXvsYChartSettings(
-                        props.chartMeta!.uniqueKey
-                    )!;
-                    settings.yLogScale = !settings.yLogScale;
-                };
-                props.onSwapAxes = () => {
-                    this.store.swapXvsYChartAxes(props.chartMeta!.uniqueKey);
-                };
-                props.plotDomain = chartInfo.plotDomain;
-                props.axisLabelX = `${chartInfo.xAttr.displayName}${
-                    settings.xLogScale ? ' (log)' : ''
-                }`;
-                props.axisLabelY = `${chartInfo.yAttr.displayName}${
-                    settings.yLogScale ? ' (log)' : ''
-                }`;
-                props.tooltip = makeDensityScatterPlotTooltip(
-                    chartInfo,
-                    settings
-                );
-                props.onValueSelection = (bounds: RectangleBounds) => {
-                    this.store.updateScatterPlotFilterByValues(
-                        props.chartMeta!.uniqueKey,
-                        bounds
-                    );
-                };
-                props.onResetSelection = () => {
-                    this.store.updateScatterPlotFilterByValues(
-                        props.chartMeta!.uniqueKey
-                    );
+                const props: any = {
```

Comment:
@7xuanlu can you fix this type?  you can see that is used to be set to Partial< > something other

Reply:
updated